libcrux-hkdf/src/hkdf.rs

```diff
@@ -8,7 +8,69 @@ pub mod hacl;
 #[cfg(feature = "hacl")]
 mod impl_hacl;
 
-use impl_hacl::{HkdfMode, HkdfSha2_256, HkdfSha2_384, HkdfSha2_512};
+pub use impl_hacl::{HkdfSha2_256, HkdfSha2_384, HkdfSha2_512};
+
+pub trait HkdfMode<const HASH_LEN: usize> {
+    /// The hash algorithm used in this HKDF mode.
+    const MODE: Algorithm;
+
+    /// HKDF extract using the `salt` and the input key material `ikm`.
+    /// The result is written to `prk`.
+    ///
+    /// Note that this function panics if `salt` or `ikm` is longer than  (2**32 - 1) bytes.
+    fn extract(prk: &mut [u8; HASH_LEN], salt: &[u8], ikm: &[u8]) -> Result<(), Error>;
+
+    /// HKDF expand using the pre-key material `prk` and `info`. The output length
+    /// is defined through the type of the `okm` parameter, that the output is written to.
+    ///
+    /// Returns nothing on success.
+    /// Returns [`Error::OkmTooLarge`] if the requested `okm_len` is too large.
+    /// Returns [`Error::ArgumentsTooLarge`] if one of `salt` or `ikm` are longer than, or
```

Comment:
```suggestion
    /// Returns [`Error::ArgumentsTooLarge`] if  `ikm` is longer than, or
```

Since there's no `salt` argment here.